imageio/plugins/spe.py

```diff
@@ -575,172 +433,426 @@ def _open(self, char_encoding="latin1", check_filesize=True, sdt_meta=True):
+            (only for files created by SDT-control)
+            Recovery time in seconds
+        comment : str
+            (only for files created by SDT-control)
+            User-entered comment. This replaces the "comments" field.
+        datetime : datetime.datetime
+            (only for files created by SDT-control)
+            Combines the "date" and "time_local" keys. The latter two plus
+            "time_utc" are removed.
+        modulation_script : str
+            (only for files created by SDT-control)
+            Laser modulation script. Replaces the "spare_4" key.
+        """
+
+        if self._file_header_ver < 3:
+            return self._metadata_pre_v3(char_encoding, sdt_control)
+        return self._metadata_post_v3()
+
+    def _metadata_pre_v3(self, char_encoding: str, sdt_control: bool) -> Dict[str, Any]:
+        """Extract metadata from SPE v2 files
+
+        Parameters
+        ----------
+        char_encoding
+            String character encoding
+        sdt_control
+            If `True`, try to decode special metadata written by the
+            SDT-control software.
 
+        Returns
+        -------
+        dict mapping metadata names to values.
+        """
+
+        m = self._parse_header(Spec.metadata, char_encoding)
+
+        nr = m.pop("NumROI", None)
+        nr = 1 if nr < 1 else nr
+        m["ROIs"] = roi_array_to_dict(m["ROIs"][:nr])
+
+        # chip sizes
+        m["chip_size"] = [m.pop(k, None) for k in ("xDimDet", "yDimDet")]
+        m["virt_chip_size"] = [m.pop(k, None) for k in ("VChipXdim", "VChipYdim")]
+        m["pre_pixels"] = [m.pop(k, None) for k in ("XPrePixels", "YPrePixels")]
+        m["post_pixels"] = [m.pop(k, None) for k in ("XPostPixels", "YPostPixels")]
+
+        # convert comments from numpy.str_ to str
+        m["comments"] = [str(c) for c in m["comments"]]
+
+        # geometric operations
+        g = []
+        f = m.pop("geometric", 0)
+        if f & 1:
+            g.append("rotate")
+        if f & 2:
+            g.append("reverse")
+        if f & 4:
+            g.append("flip")
+        m["geometric"] = g
+
+        # Make some additional information more human-readable
+        t = m["type"]
+        if 1 <= t <= len(Spec.controllers):
+            m["type"] = Spec.controllers[t - 1]
+        else:
+            m["type"] = None
+        r = m["readout_mode"]
+        if 1 <= r <= len(Spec.readout_modes):
+            m["readout_mode"] = Spec.readout_modes[r - 1]
+        else:
+            m["readout_mode"] = None
+
+        # bools
+        for k in (
+            "absorb_live",
+            "can_do_virtual_chip",
+            "threshold_min_live",
+            "threshold_max_live",
+        ):
+            m[k] = bool(m[k])
+
+        # Extract SDT-control metadata if desired
+        if sdt_control:
+            SDTControlSpec.extract_metadata(m, char_encoding)
+
+        return m
+
+    def _metadata_post_v3(self) -> Dict[str, Any]:
```

Comment:
Same here. You just moved it right?

Reply:
There were some changes. First, the old plugin read the metadata upon loading and saved it as `self._meta`. I made the code a little more compact. There is no more `"frame_shape"` metadata field since it is redundant.

Reply:
My last comment is actually about `_metadata_pre_v3`, but except for the part about `"frame_shape"` it also applies to `_metadata_post_v3`